python/ray/serve/_private/application_state.py

```diff
@@ -617,13 +619,16 @@ def get_checkpoint_data(self) -> ApplicationTargetState:
 
     def get_deployment(self, name: str) -> DeploymentInfo:
         """Get deployment info for deployment by name."""
-        return self._deployment_state_manager.get_deployment(name)
+        deployment_id = DeploymentID(name, self._name)
+        return self._deployment_state_manager.get_deployment(str(deployment_id))
```

Comment:
why convert to str here instead of using the deployment ID as the key?

Reply:
That's where the PR split is happening 😅 I'll switch to using the deployment ID as the key in the next PR, and that should make this migration complete.

Reply:
Got it!